modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/StemmerTokenFilterFactoryTests.java

```diff
@@ -103,4 +105,40 @@ public void testMultipleLanguagesThrowsException() throws IOException {
         );
         assertEquals("Invalid stemmer class specified: [english, light_english]", e.getMessage());
     }
+
+    public void testGermanAndGerman2Stemmer() throws IOException {
+        IndexVersion v = IndexVersionUtils.randomVersionBetween(random(), IndexVersions.UPGRADE_TO_LUCENE_10_0_0, IndexVersion.current());
+        Analyzer analyzer = createGermanStemmer("german", v);
+        assertAnalyzesTo(analyzer, "Buecher Bücher", new String[] { "Buch", "Buch" });
+
+        analyzer = createGermanStemmer("german2", v);
+        assertAnalyzesTo(analyzer, "Buecher Bücher", new String[] { "Buch", "Buch" });
+        assertWarnings(
+            "The 'german2' stemmer has been deprecated and folded into the 'german' Stemmer. "
+                + "Replace all usages of 'german2' with 'german'."
+        );
+    }
+
+    public Analyzer createGermanStemmer(String variant, IndexVersion v) throws IOException {
```

Comment:
super minor nit: this can probably be private? static too?

Reply:
right